### DISCUSSION TITLE
Inserting AlpineJS via Livewire gives Unhandled Promise Rejection: ReferenceError: Can't find variable

### DISCUSSION THREAD

**Discussion by joostvanhoof** [livewire/livewire#3272] [7. Bugs (archived)] [answered]
I have a Livewire component that inserts the following AlpineJS component:

```html
<div x-data="{ example: false }">
  <div x-on:click="example = ! example" :class="example ? 'class1' : 'class2'"></div>
</div>
```

This produces: `Unhandled Promise Rejection: ReferenceError: Can't find variable: example.`

The problem seems to be the `:class` part, if I remove that the error goes away. It looks like this is due to the component being inserted by Livewire, because if I put this exact snippet on a regular page (non-Livewire) it works. Adding `x-init` that sets `example: false` also doesn't work.

I'm not sure if this is a bug or a mistake on my end so hopefully someone has an idea here.

What is interesting is that if I add a component above the one I just mentioned, like so:

```html
<div x-data="{ example2: false }"></div>
```

then `example` works as expected but `example2` produces the error.

**Comment by joshhanley** [comment, marked as answer]
@joostvanhoof we fixed an issue that was causing problems with other JS libraries, can you upgrade to Livewire v2.5.5 and see if that solves the issue?

**Reply by joostvanhoof** [reply]
Yep, it works now. Thanks!

**Reply by joshhanley** [reply]
Awesome! 🙂
